directory/cli/src/host.py

```diff
@@ -0,0 +1,170 @@
+import click
+from collections import OrderedDict
+
+import list_command
+from list_command import \
+	field_with_same_name, \
+	field_with_name, \
+	hostgroup_with_host_hgid
+import ipa_wrapper_command
+import ipa_utils
+import appliance_cli.text as text
+import appliance_cli.utils
+import utils
+from exceptions import IpaRunError
+from option_transformer import OptionTransformer
+
+HOST_LIST_FIELD_CONFIGS = OrderedDict([
+	('Host name', field_with_same_name),
+	('Password', field_with_same_name),
+	('HID', field_with_same_name),
+	('Host IP', field_with_same_name),
+])
+
+HOST_SHOW_FIELD_CONFIGS = HOST_LIST_FIELD_CONFIGS.copy()
+HOST_SHOW_FIELD_CONFIGS.update([
+])
+
+# TODO add host blacklist
+HOST_BLACKLIST = []
+
+
+def add_commands(directory):
+	
+	@directory.group(help='Perform host management tasks')
+	def host():
+		pass
+
+	@host.command(help='List all hosts')
+	def list():
+		list_command.do(
+			ipa_find_command='host-find',
+			field_configs=HOST_LIST_FIELD_CONFIGS,
+			#TODO sort the sorting of the list results
+			#sort_key='HID',
+			generate_additional_data=_additional_data_for_list,
+			blacklist_key='Host name',
+			blacklist_val_array=HOST_BLACKLIST,
+		)
+
+	@host.command(help='Show detailed information on a host')		
+	#need to decide on identifier
+	@click.argument('hostname')
+	def show(hostname):
+		_validate_blacklist_hosts(hostname)
+		host_find_args = ['--hostname={}'.format(hostname)]
+		try:
+			list_command.do(
+				#or should this be 'host-show'(though it isn't in stu's code)?
```

Comment:
Yep, use `host-show`, there's an example in #10 of the command

Reply:
My bad, from the looks of things the `host-find` is used to check the host then the `generate_additional_data` handles finding the extra details using `host-show`